java-client/build.gradle.kts

```diff
@@ -248,8 +248,15 @@ tasks.withType<Jar> {
 
 publishing {
     repositories{
-        maven {
-            url = uri("${rootProject.buildDir}/repository")
+        if (version.toString().endsWith("SNAPSHOT")) {
+            maven("https://aws.oss.sonatype.org/content/repositories/snapshots/") {
+                name = "snapshotRepo"
+                credentials(PasswordCredentials::class)
```

Comment:
For my learning, how are these creds passed?

Reply:
See above, this asks for a specific environment variable that gets set by the wrapper.